openmetadata-ui/src/main/resources/ui/src/pages/RolesPage/RolesPage.component.tsx

```diff
@@ -0,0 +1,519 @@
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isLoadingPolicy, setIsLoadingPolicy] = useState<boolean>(false);
+  const [isAddingRole, setIsAddingRole] = useState<boolean>(false);
+  const [isAddingRule, setIsAddingRule] = useState<boolean>(false);
+  const [errorData, setErrorData] = useState<FormErrorData>();
+  const [isEditable, setIsEditable] = useState<boolean>(false);
+
+  const getTabs = () => {
+    return (
+      <div className="tw-mb-3 ">
+        <nav
+          className="tw-flex tw-flex-row tw-gh-tabs-container"
+          data-testid="tabs">
+          <button
+            className={`tw-pb-2 tw-px-4 tw-gh-tabs ${getActiveTabClass(
+              1,
+              currentTab
+            )}`}
+            data-testid="users"
+            onClick={() => {
+              setCurrentTab(1);
+            }}>
+            Policy
+          </button>
+          <button
+            className={`tw-pb-2 tw-px-4 tw-gh-tabs ${getActiveTabClass(
+              2,
+              currentTab
+            )}`}
+            data-testid="assets"
+            onClick={() => {
+              setCurrentTab(2);
+            }}>
+            Users
+          </button>
+        </nav>
+      </div>
+    );
+  };
+
+  const fetchLeftPanel = () => {
+    return (
+      <>
+        <div className="tw-flex tw-justify-between tw-items-center tw-mb-3 tw-border-b">
+          <h6 className="tw-heading tw-text-base">Roles</h6>
+          <NonAdminAction position="bottom" title={TITLE_FOR_NON_ADMIN_ACTION}>
+            <Button
+              className={classNames('tw-h-7 tw-px-2 tw-mb-4', {
+                'tw-opacity-40': !isAdminUser && !isAuthDisabled,
+              })}
+              data-testid="add-role"
+              size="small"
+              theme="primary"
+              variant="contained"
+              onClick={() => {
+                setErrorData(undefined);
+                setIsAddingRole(true);
+              }}>
+              <i aria-hidden="true" className="fa fa-plus" />
+            </Button>
+          </NonAdminAction>
+        </div>
+        {roles &&
+          roles.map((role) => (
+            <div
+              className={`tw-group tw-text-grey-body tw-cursor-pointer tw-text-body tw-mb-3 tw-flex tw-justify-between ${getActiveCatClass(
+                role.name,
+                currentRole?.name
+              )}`}
+              key={role.name}
+              onClick={() => setCurrentRole(role)}>
+              <p
+                className="tag-category label-category tw-self-center tw-truncate tw-w-52"
+                title={role.displayName}>
+                {role.displayName}
+              </p>
+            </div>
+          ))}
+      </>
+    );
+  };
+
+  const getPolicyRules = (rules: Array<Rule>) => {
+    if (!rules.length) {
+      return (
+        <div className="tw-text-center tw-py-5">
+          <p className="tw-text-base">No Rules Added.</p>
```

Comment:
Nice

I forgot to mention what to do when no rules are available. Good to see that we are covering this case